### ISSUE TITLE
Project search results view getting into inconsistent state

### ISSUE THREAD

**Issue by nathansobo** [atom/find-and-replace#868]
@kuychaco witnessed this on her machine. Notice how there's a rogue element in the wrong position in the results.

![image](https://cloud.githubusercontent.com/assets/1789/23685408/2713c4d6-0361-11e7-8e56-2472f63143e8.png)

No one else has been able to reproduce this, but it's concerning with respect to the new Etch implementation. It almost seems like we're losing track of a DOM node in our virtual DOM updates.

/cc @maxbrunsfeld

**Comment by Ben3eeE** [comment]
Investigated some this morning on a fresh master to include the latest etch update. I installed all packages that @kuychaco have installed so none of my results are in safe mode.

I found out that you can get into funky states and deprecations if you split the `Project Find Results` after performing a search and if you move the `Project Find Results` to another pane. I have not been able to reproduce the same behavior that @kuychaco saw.

@kuychaco Did you split the `Project Find Results` or move it to another pane after performing the search?

I'll investigate more later, need to get back to work :grinning: 

## Testing Results

Dragging the `Project Find Results` to another pane resets the scroll position but does not update the view. So results can be missing until you scroll.

**Missing search results**
![missing search result](https://cloud.githubusercontent.com/assets/1058982/23697452/21eabeb0-03e8-11e7-8973-3b8b11929708.PNG)
![missing search result dom](https://cloud.githubusercontent.com/assets/1058982/23697457/24cc4b08-03e8-11e7-8f51-4a41ce304df4.PNG)

If you open the MRU view after a split you get deprecations.

**Core deprecation**
![core deprecation](https://cloud.githubusercontent.com/assets/1058982/23697461/2afcd8a8-03e8-11e7-9c64-84b680fad25f.PNG)

**Handy gif for missing results and core deprecation**
![fnr stuff](https://cloud.githubusercontent.com/assets/1058982/23698361/8fd770f0-03eb-11e7-91c9-3b9ef142a9d3.gif)

**Weird empty pane**
![image](https://cloud.githubusercontent.com/assets/1058982/23698204/f328ae9a-03ea-11e7-9967-7c5f59ec8aa5.png)

Fooling around with the empty pane can get you to this state.

**Not searching for setPaths**
![not searching for setpaths](https://cloud.githubusercontent.com/assets/1058982/23697443/1a01219e-03e8-11e7-9bdd-8eb1174cc43e.PNG)

**Comment by BinaryMuse** [comment]
> Did you split the Project Find Results or move it to another pane after performing the search?

I don't remember but I believe this was not the case. I believe we performed one search and then a second, and it was the second search that had the odd DOM.

**Comment by Ben3eeE** [comment]
Yeah, splitting and moving only reproduces unrelated issues. I haven't had any luck reproducing the error that you were seeing in the original report.
